% 
% %% ==========================================================
% \section{Tree structure}
% \subsection{}
% \begin{frame}
% \frametitle{Question(s)}
% 
% 
% \begin{overprint}
% 
% \onslide<1>
% \[
% \textcolor{white}{ h_i(y) = h(y)\exp(\mathbf{x}_{i}^{_{'}}\cdot \beta + z_i \lambda) }
% \]
% \begin{tikzpicture} \begin{scope}[draw, line width=1pt] 
% \path (1,0.6) node (a) {$\quad$ Is $\mathbf{X}=(X_A,\ldots,X_E)'$ associated with $Y$ (adjusting for $Z$)?} ;\\
% \path (0,0) [white] node[draw] (a) {$A\cap B\cap C\cap D\cap E$};
% \path (4.5,0) [white] node (a) {: $\beta_A= \ldots = \beta_E =0$};
%  \end{scope} \end{tikzpicture}
% 
% \onslide<2>
% \[
%  h_i(y) = h(y)\exp(\mathbf{x}_{i}^{_{'}}\cdot\textcolor{blue}{\beta} + z_i \lambda) %\quad \mathrm{Cox\,\,ph\,\,regression\,\,model}
% \]
% \begin{tikzpicture} \begin{scope}[draw, line width=1pt] 
% \path (1,0.6) node (a) {$\quad$ Is $\mathbf{X}=(X_A,\ldots,X_E)'$ associated with $Y$ (adjusting for $Z$)?} ;\\
% \path (-2.5,0)  node (a) {Test};
% \path (0,0) [blue] node[draw] (a) { $A\cap B\cap C\cap D\cap E$};
% \path (3.7,0) [blue] node (a) {: $\beta_A= \ldots = \beta_E =0$};
%  \end{scope} \end{tikzpicture}
% 
% \onslide<3>
% %\[
% % h_i(y) = h(y)\exp(x_{A_i} \textcolor{blue}{\beta_A} + z_i \lambda) \quad \mathrm{Cox\,\,ph\,\,model\,\,(marginal)}
% %\]
% 
% $\beta_A= \ldots = \beta_E =0$ rejected. Which ones $\neq 0$?
% 
% \bigskip
% 
% \begin{tikzpicture} \begin{scope}[draw, line width=1pt] 
% \path (0,0) node (a) {Is $X_A$ associated with $Y$ (adjusting for $Z$)?};
% \path (4,0) [blue] node[draw] (a) {$A$};
% \path (5.2,0) [blue] node (a) {: $\beta_A=0$};
% \path (0,-.6) node (a) {Is $X_B$ associated with $Y$ (adjusting for $Z$)?};
% \path (4,-.6) node[draw] (a) {$B$};
% \path (5.2,-.6) node  (a) {: $\beta_B=0$};
% \path (6.5,-.6) node  (a) {etc.};
%  \end{scope} \end{tikzpicture}
% 
% \onslide<4>
% 
% 
% \textcolor{blue}{probe set annotations} (targeted genes/chromosomes)\\ 
% { e.g. gene g1 is measured by the probes $A$ and $B$ and belongs to chromosome c1}
% 
% \bigskip
% 
% \end{overprint}
% 
% \begin{figure}
%   \begin{tikzpicture}
%   \draw (0,-1) rectangle (11.6,4);
%   \begin{scope}[draw, line width=1pt]
% 
%       \path      (11,3.5) node (zl) {\scriptsize{age}} ;  
%       \path      (11,3) node (z) {$Z$} ;
%       \path      (6,3.5) node (xl) {\scriptsize{gene expression}} ;  
%       \path      (6,3) node (x) {$\mathbf{X}$} ;
%       \path      (1,3.5) node (yl) {\scriptsize{survival}} ;  
%       \path      (1,3) node(y) {$Y$} ;
%   
%   \path     (3,2) node[draw] (a) {$A$};
%   \path     (4.5,2) node[draw] (b) {$B$};  
%   \path      (6,2) node[draw] (c) {$C$} ;
%   \path      (7.5,2) node[draw] (d) {$D$} ;
%   \path      (9,2) node[draw] (e) {$E$} ;
%   
%   \path<2-3> [blue]    (3,2) node[draw] (a) {$A$};
%   \path<2> [blue]    (4.5,2) node[draw] (b) {$B$};  
%   \path<2>  [blue]    (6,2) node[draw] (c) {$C$} ;
%   \path<2>  [blue]    (7.5,2) node[draw] (d) {$D$} ;
%   \path<2>   [blue]   (9,2) node[draw] (e) {$E$} ;
%   
%   %data
%        \path     (11,1.25  ) node (z1) {{\scriptsize 45 yrs }};
%        \path     (11,1) node (z2) {{\scriptsize 35 yrs}};
%        \path     (11,0.75  ) node (z4) {{\scriptsize 56 yrs}};
%        \path     (11,0.5) node (z5) {{\scriptsize  32 yrs}};
%        \path     (11,0.25     ) node (z6) {{\scriptsize $\cdots$}};
%        \path     (6,1.25  ) node (a1) {{\scriptsize $5.4$}};
%        \path     (6,1) node (a2) {{\scriptsize $7.6$}};
%        \path     (6,0.75  ) node (a4) {{\scriptsize $2.1$}};
%        \path     (6,0.5) node (a5) {{\scriptsize $8.0$}};
%        \path     (6,0.25     ) node (a6) {{\scriptsize $\cdots$}};
%        \path     (7.5,1.25  ) node (a1) {{\scriptsize $7.7$}};
%        \path     (7.5,1) node (a2) {{\scriptsize $3.2$}};
%        \path     (7.5,0.75  ) node (a4) {{\scriptsize $4.9$}};
%        \path     (7.5,0.5) node (a5) {{\scriptsize $2.8$}};
%        \path     (7.5,0.25     ) node (a6) {{\scriptsize $\cdots$}};
%        \path     (9,1.25  ) node (a1) {{\scriptsize $2.1$}};
%        \path     (9,1) node (a2) {{\scriptsize $6.4$}};
%        \path     (9,0.75  ) node (a4) {{\scriptsize $8.5$}};
%        \path     (9,0.5) node (a5) {{\scriptsize $7.2$}};
%        \path     (9,0.25     ) node (a6) {{\scriptsize $\cdots$}};
%        \path     (4.5,1.25  ) node (a1) {{\scriptsize $5.9$}};
%        \path     (4.5,1) node (a2) {{\scriptsize $9.8$}};
%        \path     (4.5,0.75  ) node (a4) {{\scriptsize $8.2$}};
%        \path     (4.5,0.5) node (a5) {{\scriptsize $6.3$}};
%        \path     (4.5,0.25     ) node (a6) {{\scriptsize $\cdots$}};
%         \path     (3,1.25  ) node (a1) {{\scriptsize $8.2$}};
%        \path     (3,1) node (a2) {{\scriptsize $3.6$}};
%        \path     (3,0.75  ) node (a4) {{\scriptsize $5.1$}};
%        \path     (3,0.5) node (a5) {{\scriptsize $5.8$}};
%        \path     (3,0.25     ) node (a6) {{\scriptsize $\cdots$}};
%        \path     (1,1.25  ) node (a1) {{\scriptsize $5.8$}};
%        \path     (1,1) node (a2) {{\scriptsize $8.1$}};
%        \path     (1,0.75  ) node (a4) {{\scriptsize $3.9$}};
%        \path     (1,0.5) node (a5) {{\scriptsize $10.2+$}};
%        \path     (1,0.25     ) node (a6) {{\scriptsize $\cdots$}};
%  
%   %biological info
%                 \path<4>     (1,-0.25     ) [blue] node (biog) {{\scriptsize gene}};
%                 \path<4>     (1,-0.5     ) [blue] node (bioc) {{\scriptsize chromosome}};
%                      \path<4>     (3,-0.25     )[blue] node (bioc) {{\scriptsize g1}};
%                      \path<4>     (4.5,-0.25     )[blue] node (bioc) {{\scriptsize g1}};
%                      \path<4>     (6,-0.25     )[blue] node (bioc) {{\scriptsize g2}};
%                      \path<4>     (7.5,-0.25     )[blue] node (bioc) {{\scriptsize g3}};
%                      \path<4>     (9,-0.25     )[blue] node (bioc) {{\scriptsize g4}};
%                      \path<4>     (3,-0.5     )[blue] node (bioc) {{\scriptsize c1}};
%                      \path<4>     (4.5,-0.5     )[blue] node (bioc) {{\scriptsize c1}};
%                      \path<4>     (6,-0.5     )[blue] node (bioc) {{\scriptsize c1}};
%                      \path<4>     (7.5,-0.5     )[blue] node (bioc) {{\scriptsize c2}};
%                      \path<4>     (9,-0.5     )[blue] node (bioc) {{\scriptsize c2}};
%   \end{scope}
%   \end{tikzpicture}
% \end{figure}
% 
% \end{frame}
% %% ==========================
% \subsection{A priori}
% \begin{frame}
% \frametitle{Domain-based tree (A priori structure) }
% 
% 
% \begin{overprint}
% 
% \onslide<1>
% 
% \bigskip
% \bigskip
% 
% Build up the tree
% 
% \onslide<2>
% 
% Add a new hypothesis\\
% 
% \begin{tikzpicture} \begin{scope}[draw, line width=1pt] 
% \path (0,0) [blue] node[draw] (a) {$A \cap B$};
% \path (2,0) [blue] node (a) {: $\beta_A=\beta_B=0$};
%  \end{scope} \end{tikzpicture}
% 
% \onslide<3>
% 
% 
% top = general question,  $\,\,\downarrow\, $ = more specific follow-up questions
% 
% \bigskip
% 
% Logically related hypotheses: e.g. $AB = A\cap B$ implies $A$ and $B$
% 
% 
% \end{overprint}
% 
% \begin{figure}
%   \begin{tikzpicture}
%   \draw (0,-1) rectangle (11.6,4);
%   \begin{scope}[draw, line width=1pt]
%   \path<1->     (1,0) node[draw] (a) {$A$};
%   \path<1->     (3,0) node[draw] (b) {$B$};
%   \path<2>     (1.9,1) [blue] node[draw] (ab) {$A\cap B$} ;
%   \path<1->      (4,1)  node[draw] (c) {$C$} ;
%   \path<1->      (6,1)  node[draw] (d) {$D$} ;
%   \path<1->      (8,1)  node[draw] (e) {$E$} ;
%   \path<3->      (4,1)  node[draw] (c) {$C$} ;
%   \path<3->      (6,1)  node[draw] (d) {$D$} ;
%   \path<3->      (8,1)  node[draw] (e) {$E$} ;
%   \path<1->     (10.5,0)  node {{\scriptsize probe level}};
%   \path<1->     (10.5,1) node {{\scriptsize gene level}};
%   \path<1->     (10,2)  node {{\scriptsize chromosome level}};
%   \path<3>     (5,3) node[draw] (top) {$ABCDE$};
%   \path<3>     (3,2)  node[draw] (abc) {$ABC$} ;
%   \path<3>     (7,2) node[draw] (de) {$DE$} ;
%   \path<3>     (1.9,1) node[draw] (ab) {$AB$} ;
%   \path<1->     (10.6,3) node {{\scriptsize top level}};
%   \end{scope}
%   \begin{scope}[thick, shorten >= 2pt,shorten <= 2pt, latex-]
%     \draw<3-> (abc.north) -- (top.south);
%     \draw<3-> (de.north) -- (top.south);
%     \draw<3-> (ab.north) -- (abc.south);
%     \draw<3-> (c.north) -- (abc.south);
%     \draw<2-> (b.north) -- (ab.south);
%     \draw<2-> (a.north) -- (ab.south);
%     \draw<3-> (e.north) -- (de.south);
%     \draw<3-> (d.north) -- (de.south);
%   \end{scope}
%   \end{tikzpicture}
% \end{figure}
% 
% \end{frame}
% %% ==========================
% \subsection{Data driven}
% 
% \subsection{}
% \begin{frame}[fragile]
% \frametitle{Data Driven structure}
% 
% Is it possibile to get a structure when a natural one is missing?
% 
% \bigskip
% 
% Hierarchical clustering with some $\mathbf{X}$-based (e.g. gene expression) distance matrix 		
%   
%   \bigskip
%   
% \begin{figure}
% \begin{center}
%   \includegraphics[height=.4\textwidth,width=.4\textwidth]{hclust}
% \end{center}
% \end{figure}
% 
% \end{frame}
% %% ========================
% \begin{frame}
% \frametitle{Correlation-driven tree (intuition)}
% 
% \begin{figure}
%   \includegraphics[height=6cm]{plots}
% \end{figure}
% 
% \end{frame}
% %%  =========================
% \subsection{}
% \begin{frame}[fragile]
% \frametitle{Correlation-driven tree (intuition)}
% 
% \begin{columns}
% 
% \column{.65\textwidth}
% 
% \begin{overprint}
% 
% \onslide<1-4> 
% $X_A=\alpha_A+\beta_AY +\epsilon_A$,\qquad $X_B=\alpha_B+\beta_BY +\epsilon_B$\\
% r.v. $\epsilon_A$, $\epsilon_B$ and $Y$  independent each other
% 
% \end{overprint}
% 
%   \begin{tikzpicture}
%   \draw (0,-1) rectangle (6.8,4);
%   \begin{scope}[draw, line width=1.25pt]
% 
% \draw<1-4> (1.5,2.5) node[minimum size=1.5cm,draw,circle] (xe) {$X_E$};
% \draw<1-4> (1.5,.5) node[minimum size=1.5cm,draw,circle] (xd) {$X_D$};  
% \draw<1-4> (3,1.5) node[minimum size=1.5cm,draw,circle] (xc) {$X_C$};
% \draw<1-4> (4.5,.5) node[minimum size=1.5cm,draw,circle] (xa) {$X_A$};
% \draw<1-4> (4.5,2.5) node[minimum size=1.5cm,draw,circle] (xb) {$X_B$};
% \draw<2> (6,1.5) [blue] node[minimum size=1.5cm,draw,circle] {$Y$};
% \draw<3-4> (4.6,1.5) [blue] node[minimum size=1.5cm,draw,circle] {$Y$};
% 
%   \end{scope}
%     \begin{scope}[thick, shorten >= 2pt,shorten <= 2pt]
%     \draw<3-4>[<->] [blue] (xa.east) to [bend right=45] (xb.east);
% %    \draw<3-4>[<->, dotted] [blue] (xc.south) to [bend right=45] (xa);
% %    \draw<3-4>[<->, dotted] [blue] (xc.north) to [bend left=45] (xb);
% 
%    \end{scope}
%   \end{tikzpicture}
%   
% \column{.35\textwidth}
% 
% \begin{overprint}
% 
% \onslide<1> 
% \bb{}{
% %\begin{center}
% %suppose all $X$s uncorrelated
% %\end{center}
% }\eb
% 
% \onslide<2> 
% \bb{}{
% \begin{center}
% if $\beta_A=\beta_B=0$ \\(i.e. $Y$ not associated),\\ $\rightarrow \rho(X_A,X_B)=0$
% \end{center}
% }\eb
% 
% \onslide<3> 
% \bb{}{
% \begin{center}
% if $\beta_A>0,\ \beta_B>0$ \\(i.e. $Y$ associated), $\rightarrow \rho(X_A,X_B)>0$ \\ 
% 
% \textcolor{blue}{spurious associations}  (same as confounding)
% \end{center}
% }\eb
% 
% \onslide<4> 
% \bb{}{
% \begin{center}
% hierarchical clustering based on absolute correlation among $X$s ($H_1$s are more likely to be clustered together)
% \begin{figure}
%   \includegraphics[height=.9\textwidth,width=.9\textwidth]{hclust}
% \end{figure}
% \end{center}
% }\eb
% 
% \end{overprint}
%   
% \end{columns}
%   
% \end{frame}
% 
% %% ==============
% %% =================
% \begin{frame}[fragile]
% \frametitle{ Data Driven structure}
% 
% \begin{columns}
% \column{0.3\textwidth}
% 
% \begin{block}{
% hierarchical clustering } 
% \includegraphics[scale=.3]{hclust}
% \end{block}
% 
% \column{0.01\textwidth}
% 
% \line(0,1){180}
% 
% 
% \column{0.7\textwidth}
% \begin{block}{
% data-driven tree } 
% 
%   \begin{tikzpicture}
%   \begin{scope}[draw, line width=1pt]
%   \path<1>          (5,3) node[draw] (top) {$ABCDE$} ;
%   \path<1>     (3,2)  node[draw] (abc) {$ABC$} ;
%   \path<1>     (7,2) node[draw] (de) {$DE$} ;
%   \path<1>     (1.9,1) node[draw] (ab) {$AB$} ;
%   \path<1>     (4,1) node[draw] (c) {$C$} ;
%   \path          (6,1) node[draw] (d) {$D$} ;
%   \path           (8,1) node[draw] (e) {$E$} ;
%   \path     (1,0) node[draw] (a) {$A$};
%   \path            (3,0) node[draw] (b) {$B$};
%    \end{scope}
%   \begin{scope}[thick, shorten >= 2pt,shorten <= 2pt, latex-]
%     \draw         (abc.north) -- (top.south);
%     \draw         (de.north) -- (top.south);
%     \draw         (ab.north) -- (abc.south);
%     \draw (c.north) -- (abc.south);
%     \draw (b.north) -- (ab.south);
%     \draw (a.north) -- (ab.south);
%     \draw (b.north) -- (ab.south);
%     \draw (e.north) -- (de.south);
%     \draw (d.north) -- (de.south);
%   \end{scope}
%   \end{tikzpicture}
% \end{block}
% 
% \end{columns}
% 
% 
% \end{frame}
% %% =======================
% 

%% ==========================INHERITANCE
% \section{Inheritance procedure}
\subsection{}

\begin{frame}
\frametitle{Meinshausen's procedure (BMK 2008)}

\begin{overprint}

 \onslide<1>
 
 \bb{}{
   start from the top: \\
   test $ABCDE$ at level $\alpha$}\eb
  \onslide<2>
   \bb{}{
   suppose $p_{ABCDE}\leq \alpha$, \\
   test $ABC$ at $\frac{3}{5}\alpha$ and $DE$ at $\frac{2}{5} \alpha$}\eb
   \onslide<3>
    \bb{}{
   suppose $p_{ABC}\leq \frac{3}{5}\alpha$ and $p_{DE}>\frac{2}{5} \alpha$}\eb
   \onslide<4>
    \bb{}{
   test $AB$ at $\frac{2}{5}\alpha$ and $C$ at $\frac{1}{5}\alpha$}\eb
   \onslide<5>
    \bb{}{
   suppose $p_{AB}\leq \frac{2}{5}\alpha$ and $p_{C}> \frac{1}{5}\alpha$, \\
   test $A$ at  $\frac{1}{5}\alpha$ and $B$ at $\frac{1}{5}\alpha$}\eb
   \onslide<6>
    \bb{}{
   suppose $p_{A}\leq \frac{1}{5}\alpha$ and $\frac{1}{5}\alpha < p_{B}< \frac{2}{5}\alpha$\\
   STOP?}\eb
  \onslide<7>
      \bb{}{
    \textcolor{green}{Shaffer's improvement:} if $A\cap B$ is a correct rejection, \\ at least one hypothesis is false: test $A$ and $B$ at level $\frac{2}{5}\alpha$
}\eb
  \onslide<8>
      \bb{}{
    reject A and B. \\
    STOP!}\eb
\end{overprint}

\begin{figure}
  \begin{tikzpicture}
  \draw (0,-1) rectangle (10,4);
  \begin{scope}[draw, line width=1pt]
    \path<1>   [blue]       (5,3) node[draw] (top) {$ABCDE$} ;
  \path<1>[blue]    (top.east)   node {$\quad \alpha$} ;
  \path<2>[blue] (5,3) node[draw] (top) {$ABCDE$} ;
  \path<2->[red] (5,3) node[draw, cross out] (top) {$ABCDE$} ;
  \path<2->[red] (5,3) node[draw] (top) {$ABCDE$} ;
  \path<1-2>     (3,2)  node[draw] (abc) {$ABC$} ;
  \path<2>[blue] (3,2)  node[draw] (abc) {$ABC$} ;
  \path<2>[blue] (abc.east)  node {$\ \quad \alpha \frac{3}{5}$} ;
  \path<3->[red] (3,2) node[draw, cross out] (abc) {$ABC$} ;
  \path<3->[red] (3,2) node[draw] (abc) {$ABC$} ;
  \path<1-2>     (7,2) node[draw] (de) {$DE$} ;
  \path<2->[blue] (7,2) node[draw] (de) {$DE$} ;
  \path<2->[blue] (de.east) node {$\ \quad \alpha \frac{2}{5}$} ;
  \path<1-3>     (1.9,1) node[draw] (ab) {$AB$} ;
  \path<4>[blue] (1.9,1) node[draw] (ab) {$AB$} ;
  \path<4>[blue] (ab.east) node {$\ \quad  \alpha \frac{2}{5}$} ;
  \path<5->[red] (1.9,1) node[draw, cross out] {$AB$} ;
  \path<5->[red] (1.9,1) node[draw] {$AB$} ;
  \path<1-3>     (4,1) node[draw] (c) {$C$} ;
  \path<4->[blue] (4,1) node[draw] (c) {$C$} ;
  \path<4->[blue] (c.east) node {$\ \quad \alpha \frac{1}{5}$} ;
  \path          (6,1) node[draw] (d) {$D$} ;
  \path           (8,1) node[draw] (e) {$E$} ;
  \path     (1,0) node[draw] (a) {$A$};
  \path<5>[blue] (1,0) node[draw] (a) {$A$};
  \path<6>[red]  (1,0) node[draw] (a) {$A$};
  \path<6>[red]  (1,0) node[draw, cross out] (a) {$A$};
  \path<7>[blue] (1,0) node[draw] (a) {$A$};
  \path<8>[red]  (1,0) node[draw] (a) {$A$};
  \path<8>[red]  (1,0) node[draw, cross out] (a) {$A$};
  \path<5>[blue] (a.east)node {$\ \quad \alpha \frac{1}{5}$} ;
  \path<7>[green](a.east)node[draw] {$\ \quad \alpha \frac{2}{5}$} ;
  \path            (3,0) node[draw] (b) {$B$};
  \path<5->[blue]   (3,0) node[draw] (b) {$B$};
  \path<5-6>[blue]  (b.east) node {$\ \quad \alpha \frac{1}{5}$} ;
  \path<7>[green]    (b.east)node[draw] {$\ \quad  \alpha \frac{2}{5}$} ;
  \path<8->[red]    (3,0) node[draw] (b) {$B$};
  \path<8->[red]    (3,0) node[draw, cross out] (b) {$B$};
  \end{scope}
  \begin{scope}[thick, shorten >= 2pt,shorten <= 2pt, latex-]
    \draw         (abc.north) -- (top.south);
    \draw<2->[red] (abc.north) -- (top.south);
    \draw         (de.north) -- (top.south);
    \draw<2>[red] (de.north) -- (top.south);
    \draw         (ab.north) -- (abc.south);
    \draw<4->[red] (ab.north) -- (abc.south);
    \draw (c.north) -- (abc.south);
    \draw<4>[red] (c.north) -- (abc.south);
    \draw (b.north) -- (ab.south);
    \draw (a.north) -- (ab.south);
    \draw (b.north) -- (ab.south);
    \draw<5-6, 8>[red] (a.north) -- (ab.south);
    \draw<5>[red] (b.north) -- (ab.south);
        \draw<7>[green] (a.north) -- (ab.south);
    \draw<7>[green] (b.north) -- (ab.south);
        \draw<8>[red] (b.north) -- (ab.south);
    \draw<4->[red] (d.north) -- (de.south);
    \draw (e.north) -- (de.south);
    \draw (d.north) -- (de.south);
  \end{scope}
  \end{tikzpicture}
\end{figure}
\end{frame}
%%% ==========================

\subsection{}
\begin{frame}
\frametitle{Inheritance procedure (Goeman \& Finos 2009)}

\begin{overprint}
 \onslide<1-6>\bb{}{
   Perform Meinshausen's procedure}\eb
  \onslide<7>
    \bb{}{All leaf nodes in $AB$ are rejected }\eb
      \onslide<8>\bb{}{
    $\frac{2}{5}\alpha$ from $AB$ is \textcolor{green}{inherited} to $C$ (i.e. the closest relative)\\
    test $C$ at $(\frac{1}{5} + \textcolor{green}{\frac{2}{5}})\alpha$}\eb
      \onslide<9>\bb{}{
   suppose $p_C\leq \frac{3}{5}\alpha$}\eb
      \onslide<10>\bb{}{
$\frac{3}{5}\alpha$ from $C$ is \textcolor{green}{inherited} to $DE$}\eb
\end{overprint}

\begin{figure}
  \begin{tikzpicture}
  \draw (0,-1) rectangle (10,4);
  \begin{scope}[draw, line width=1pt]
  \path          (5,3) node[draw] (top) {$ABCDE$} ;
  \path<2>[blue]    (top.east)   node {$\quad \alpha$} ;
  \path<2>[blue] (5,3) node[draw] (top) {$ABCDE$} ;
  \path<3->[red] (5,3) node[draw] (top) {$ABCDE$} ;
  \path<3->[red] (5,3) node[draw,cross out] (top) {$ABCDE$} ;
  \path<1-2>     (3,2)  node[draw] (abc) {$ABC$} ;
  \path<3>[blue] (3,2)  node[draw] (abc) {$ABC$} ;
  \path<3>[blue] (abc.east)node {$\ \quad \alpha \frac{3}{5}$} ;
  \path<4->[red] (3,2) node[draw] (abc) {$ABC$} ;
  \path<4->[red] (3,2) node[draw,cross out] (abc) {$ABC$} ;
  \path<1-2>     (7,2) node[draw] (de) {$DE$} ;
  \path<3->[blue] (7,2) node[draw] (de) {$DE$} ;
  \path<3-9>[blue] (de.east)node {$\ \quad \alpha \frac{2}{5}$} ;
  \path<10->[green] (de.east)node[draw] {$\quad \qquad \alpha \frac{5}{5}$} ;
  \path<1-4>     (1.9,1) node[draw] (ab) {$AB$} ;
  \path<5>[blue] (1.9,1) node[draw] (ab) {$AB$} ;
  \path<5>[blue] (ab.east)node {$\ \quad \alpha \frac{2}{5}$} ;
  \path<6->[red] (1.9,1) node[draw] (ab) {$AB$} ;
  \path<6->[red] (1.9,1) node[draw,cross out] (ab) {$AB$} ;
  \path<1-4>     (4,1) node[draw] (c) {$C$} ;
  \path<5->[blue] (4,1) node[draw] (c) {$C$} ;
  \path<9->[red] (4,1) node[draw] (c) {$C$} ;
  \path<9->[red] (4,1) node[draw,cross out] (c) {$C$} ;
  \path<5-7>[blue] (c.east)node {$\ \quad \alpha \frac{1}{5}$} ;
  \path<8>[green] (c.east)node[draw] {$\ \quad \alpha \frac{3}{5}$} ;
  \path          (6,1) node[draw] (d) {$D$} ;
  \path           (8,1) node[draw] (e) {$E$} ;
  \path     (1,0) node[draw] (a) {$A$};
  \path<6>[blue] (1,0) node[draw] (a) {$A$};
  \path<7->[red]  (1,0) node[draw] (a) {$A$};
  \path<7->[red]  (1,0) node[draw,cross out] (a) {$A$};
  \path<6>[blue] (a.east)node {$\ \quad \alpha \frac{2}{5}$} ;
  \path            (3,0) node[draw] (b) {$B$};
  \path<6>[blue]   (3,0) node[draw] (b) {$B$};
  \path<7->[red]    (3,0) node[draw] (b) {$B$};
  \path<7->[red]    (3,0) node[draw,cross out] (b) {$B$};
    \path<6>[blue] (b.east)node {$\ \quad \alpha \frac{2}{5}$} ;
  \end{scope}
  \begin{scope}[thick, shorten >= 2pt,shorten <= 2pt, latex-]
    \draw<1-9>     (abc.north) -- (top.south);
    \draw<3-9>[red] (abc.north) -- (top.south);
    \draw<10>[green](top.south) -- (abc.north);
    \draw<10>[green](de.north) --(top.south) ;
    \draw<1-9>         (de.north) -- (top.south);
    \draw<3>[red] (de.north) -- (top.south);
    \draw<1-7>         (ab.north) -- (abc.south);
    \draw<5-7, 9->[red] (ab.north) -- (abc.south);
    \draw<8>[green] (abc.south) -- (ab.north);
    \draw<5>[red] (c.north) -- (abc.south);
    \draw<1-4> (c.north) -- (abc.south);
    \draw<6-7> (c.north) -- (abc.south);
    \draw<8>[green] (c.north) --(abc.south) ;
     \draw<9>[red] (c.north) --(abc.south) ;
    \draw<10>[green] (abc.south) --(c.north) ;
    \draw <1-7>(b.north) -- (ab.south);
    \draw <1-7>(a.north) -- (ab.south);
    \draw<1-7> (a.north) -- (ab.south);
    \draw<1-7> (b.north) -- (ab.south);
    \draw<6-7, 9->[red] (a.north) -- (ab.south);
    \draw<6-7, 9->[red] (b.north) -- (ab.south);
    \draw<8>[green] (ab.south) -- (a.north);
    \draw<8>[green] (ab.south) -- (b.north);
    \draw (d.north) -- (de.south);
    \draw (e.north) -- (de.south);
  \end{scope}
  \end{tikzpicture}
\end{figure}

\end{frame}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Application: Genomics}

\begin{frame}
\frametitle{Microarray-based comparative genomic hybridization (aCGH) Modena et al., 2006 }
  \bb{Data}
    \bi
    \item 2 samples: \\
{\it infratentorial} tumors (14 patients) versus \\ {\it supratentorial} tumors (8 patients). 
 \item chromosome 9 (147 probes)
    \ei
  \eb
  \bb{Inference}
    \begin{itemize}
      \item for each probe (univariate): two sample t-test
      \item The signal is very weak and sparse among probes: \\
      Holm ($\alpha = .05$): no rejections
    \end{itemize}
  \eb
\end{frame}


%%%%%%%%%%%%%%%%%%%%%%%%
\bfr{Results: Inheritance}
\includegraphics[scale=.33]{Modena_ch9}

Significant differences ($\alpha = .05$):
\bb{}
  \rbf{-$\circ$} arm: p 
    \\\rbf{---$\circ$} band: p22.1 and p.21.3
    \\\rbf{-----$\circ$} gene: CTD 2097k16
\eb
\end{frame}
%%================================
\subsection{}
\begin{frame}[fragile]
\frametitle{Software (bioconductor.org)}

\bb{Inheritance procedure}
R package \emph{globaltest}\\
Authors: Goeman \& Finos
\eb
  
\begin{verbatim}
> library(globaltest)
> inheritance()

\end{verbatim}  

\end{frame}

% %==========================DISCUSSION
% %\section{Discussion}
% \subsection{}
% \bfr{Take-home message}
% 
% \bb{Inheritance procedure}
%   \bi 
%     \item Allows inference on tree-structured hypotheses
%     \item Improves Meinshausen
%     \item Available in the R package \emph{globaltest}
% 	\ei
% \eb
% 
% \bb{Future work}
%     \bi 
%     \item Extend to general graphs
%     \item Take into account the dependence between tests
%      \ei
% \eb
% 
% \end{frame}